src/sentry/profiles/flamegraph.py

```diff
@@ -142,51 +141,66 @@ def get_profile_ids_with_spans(
     project_id: int,
     params: ParamsType,
     span_group: str,
-    backend: str,
-    query: Optional[str] = None,
 ):
-    data = query_profiles_data(
-        params,
-        Referrer.API_STARFISH_PROFILE_FLAMEGRAPH.value,
-        selected_columns=["id", "profile.id"],
-        query=query,
-        additional_conditions=[
-            Condition(Function("has", [Column("spans.group"), span_group]), Op.EQ, 1)
+    query = Query(
+        match=Entity(EntityKey.Spans.value),
+        select=[
+            Column("profile_id"),
+            Function(
+                "groupArray",
+                parameters=[
+                    Function(
+                        "tuple",
+                        [
+                            Column("start_timestamp"),
+                            Column("start_ms"),
+                            Column("end_timestamp"),
+                            Column("end_ms"),
+                        ],
+                    )
+                ],
+                alias="intervals",
+            ),
+        ],
+        groupby=[
+            Column("profile_id"),
```

Comment:
This group by can also potentially cause OOMs if we're not careful here.

Reply:
Here the number of "aggregated" row per single profile is directly tied to the number of spans within the transaction profiled with a given `span group`.

Not sure if this is exactly your concern but in that case a `groupArray(LIMIT)` as suggested in the comment above should address this too.

Or maybe you meant something else? in that case what are you suggesting?

Reply:
I'm not 100% sure about this because of the lack of order by in this query but normally when you group by a column, clickhouse has to keep **all** possible values in memory while computing the aggregation for each possible value.